pkg/querier/tracing_set.go

```diff
@@ -0,0 +1,95 @@
+package querier
+
+import (
+	"context"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/storage"
+	"github.com/prometheus/prometheus/tsdb/chunkenc"
+)
+
+func newTracingSet(ctx context.Context, op string, set storage.SeriesSet) storage.SeriesSet {
+	span, _ := opentracing.StartSpanFromContext(ctx, op)
+	return &tracingSet{
+		s:    set,
+		span: span,
+	}
+}
+
+type tracingSet struct {
+	s    storage.SeriesSet
+	span opentracing.Span
+
+	series  int
+	samples int
+}
+
+func (t *tracingSet) Next() bool {
+	if t.s.Next() {
+		t.series++
+		return true
+	}
+
+	if t.span != nil {
+		t.span.LogKV("series", t.series, "samples", t.samples)
+		t.span.Finish()
+		// Nil to avoid double closing.
+		t.span = nil
+	}
+	return false
+}
+
+func (t *tracingSet) At() storage.Series {
+	s := t.s.At()
+	if t.span != nil && t.series%100 == 0 {
```

Comment:
A bit scared about this % 100. 100K series (which is still reasonable) will end up into 1K logs. A %1000 could be enough.